packages/react/src/utils/versionSelector.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * This version selector is a bit complicated, so here is the flow:
+ * 1. Use the clerkJSVersion prop on the provider
+ * 2. Use the exact `@clerk/clerk-js` version if it is a `@staging`/`snapshot` prerelease for `@clerk/clerk-react`
+ * 3. Use the prerelease tag of `@clerk/clerk-react`
+ * 4. Fallback to the major version of `@clerk/clerk-react`
+ * @param clerkJSVersion - The optional clerkJSVersion prop on the provider
+ * @returns The npm tag, version or major version to use
+ */
+export const versionSelector = (clerkJSVersion: string | undefined) => {
+  if (clerkJSVersion) {
+    return clerkJSVersion;
+  }
+
+  const prereleaseTag = getPrereleaseTag(PACKAGE_VERSION);
+  if (prereleaseTag) {
+    if (prereleaseTag === 'staging' || prereleaseTag === 'snapshot') {
```

Comment:
I definitely agree with the proposed change here for the snapshot environment, but can you please expand a little on why we need the same for the staging environment as well?

Reply:
`staging` (which gets released after PRs are merged into main branch) is also a tagged release. If you want to test the main branch in a certain state, you'll need to make sure that also the clerk-js version is the correct one.

Basically `staging` has the same problem as `snapshot`

Reply:
I am still slightly confused :) Why do we consider staging to be a tagged release? What would be the use case of testing a staging release using a specific staging version instead of using the most recent `@staging` package?

Snapshot and staging releases are still different in my mind - of course, I'm very open to discussion. Snapshot releases need to be bound with their corresponding `clerk-js` version because we can have multiple PRs open, generating snapshot releases simultaneously.

Staging releases behave differently, though. Only a single branch (`main`) triggers staging releases, so even if multiple staging releases happen each day, this always happens sequentially. Moreover, the `main` branch should always be deployable, meaning that if you test a staging version that is not the most recent one, you are not testing the code that will actually be deployed.

The main reason to have a staging environment is to catch and address bugs before they reach production. If the staging environment doesn't mirror production, it might not replicate the issues that could arise in the production environment. In our case that'd be:
- hotloading using a tag (eg `@staging`) and not a specific version
- making sure that the SDKs (eg `clerk-react`) works with the most recent `clerk-js` release

Reply:
I also agree with @nikosdouvlis here. I would keep this change only for the `snapshot` env.

Reply:
@nikosdouvlis @anagstef Made the change :)